src/de/serde.rs

```diff
@@ -4,7 +4,7 @@ use serde::de::{
     self,
     Deserialize,
     DeserializeSeed,
-    Deserializer,
+    Deserializer as SerdeDeserializer,
```

Comment:
I think this rename needs to occur because the original name conflicts with the newly-named `Deserialize` struct, but the import needs to remain as-is for [this line](https://github.com/mongodb/bson-rust/blob/ed1740acde6b1f21007af8872d275404e6f415e1/src/decoder/serde.rs#L494), but let me know if there's a better workaround.

Reply:
Since you don't need to use the name `SerdeDeserializer` at all, I think you can actually use `Deserializer as _`, which will include it for the purposes of values that implement that trait but won't put any name into scope.